uefi/src/proto/network/http.rs

```diff
@@ -0,0 +1,370 @@
+#![cfg(target_os = "uefi")]
+
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! HTTP Protocol
+
+extern crate alloc;
+use alloc::string::String;
+use alloc::vec;
+use alloc::vec::Vec;
+use core::ffi::{c_char, c_void, CStr};
+use core::ptr;
+use log::debug;
+
+use uefi::boot::ScopedProtocol;
+use uefi::prelude::*;
+use uefi::proto::unsafe_protocol;
+use uefi_raw::protocol::driver::ServiceBindingProtocol;
+use uefi_raw::protocol::network::http::{
+    HttpAccessPoint, HttpConfigData, HttpHeader, HttpMessage, HttpMethod, HttpProtocol,
+    HttpRequestData, HttpResponseData, HttpStatusCode, HttpToken, HttpV4AccessPoint, HttpVersion,
+};
+
+/// HTTP Protocol
```

Comment:
Please adjust this to our new (soon to be merged) style/doc guidline

https://github.com/rust-osdev/uefi-rs/pull/1612/commits/bc0d3cc72bb0f22368e8449f546634c407148da1

```suggestion
/// HTTP [`Protocol`]. <optional a nice short one sentence long description>
///
/// [`Protocol`]: uefi::proto::Protocol
```